torch_xla/experimental/pjrt_backend.py

```diff
@@ -19,8 +20,17 @@ def _pjrt_rendezvous_handler(url: str,
   if not master_ip:
     master_ip = tpu.discover_master_worker_ip() if xr.device_type(
     ) == 'TPU' else 'localhost'
-
-  master_port = xu.getenv_as('MASTER_PORT', int, 12355)
+  # Workaround for Neuron to prevent socket in use error with torchrun
+  # MASTER_PORT is set by torchrun and will clash with the port here
```

Comment:
I'm not sure I understand the bug here. Which two processes are trying to use this port? Does `torchrun` itself listen on `MASTER_PORT` or start a process that does? This implementation is nearly identical to the built-in `env://` `init_method`, except for some magic around discovering the world size and rank from the runtime. We also handle the multithreaded edge case for TPU v2 and v3, which AFAIK is not relevant to Neuron

Reply:
Yeah torchrun has a default MASTER_PORT so in the case here it will try to listen on the same MASTER_PORT and get a socket in use error

Reply:
Got it. I didn't realize `torchrun` itself actually listens on the MASTER_PORT -- I thought it just told the master which port to listen on. This is a bug on our side then.

I found the implementation of the `env://` `init_method`, which [sets `is_master=False` when using torchrun](https://github.com/pytorch/pytorch/blob/a14d99bb6c16233f9c305c089d1bdcdd5b8ea5ac/torch/distributed/rendezvous.py#L166-L174) and wraps `TCPStore` in a `PrefixStore`. I can copy that logic in a follow-up, or you can add it here.

Reply:
Yeah could you copy that

Reply:
Sounds good. I'll follow up with a PR that disables `is_master` and uses torchrun's MASTER_PORT instead.